contracts/HomoraIBSwap.sol

```diff
@@ -0,0 +1,226 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.7.0;
+
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/token/ERC20/IERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/token/ERC20/SafeERC20.sol";
+import "OpenZeppelin/openzeppelin-contracts@3.4.0/contracts/math/SafeMath.sol";
+import "./Governable.sol";
+import "../interfaces/IUniswapV2Router02.sol";
+import "../interfaces/ISafeBox.sol";
+import "../interfaces/ISafeBoxETH.sol";
+import "../interfaces/ICErc20.sol";
+import "../interfaces/ICyToken.sol";
+
+/// @title Homora IBToken Swap
+/// @author Sawit Trisirisatayawong (@tansawit)
+contract HomoraIBSwap is Governable {
+    using SafeERC20 for IERC20;
+    using SafeMath for uint256;
+
+    mapping(address => bool) public isIBToken;
+
+    address public constant IBETHV2 =
+        0xeEa3311250FE4c3268F8E684f7C87A82fF183Ec1;
+
+    address UNISWAP_ROUTER_ADDRESS = 0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+    IUniswapV2Router02 public uniswapRouter =
+        IUniswapV2Router02(UNISWAP_ROUTER_ADDRESS);
+
+    constructor() public {
+        __Governable__init();
+        isIBToken[IBETHV2] = true;
+    }
+
+    /// @notice add a list of token addresses as supported ibToken
+    /// @param tokens list of symbols to support
+    function addIBTokens(address[] memory tokens) external onlyGov {
+        for (uint256 idx = 0; idx < tokens.length; idx++) {
+            if (!isIBToken[tokens[idx]] && tokens[idx] != IBETHV2) {
+                isIBToken[tokens[idx]] = true;
+
+                SafeBox safebox = SafeBox(tokens[idx]);
+                IERC20(safebox.uToken()).safeApprove(
+                    UNISWAP_ROUTER_ADDRESS,
+                    uint256(-1)
+                );
+                IERC20(safebox.uToken()).safeApprove(
+                    address(safebox),
+                    uint256(-1)
+                );
+            }
+        }
+    }
+
+    function getPath(address tokenIn, address tokenOut)
+        internal
+        view
+        returns (address[] memory)
+    {
+        address[] memory path;
+        address underlyingTokenIn;
+        address underlyingTokenOut;
+
+        if (tokenIn == IBETHV2) {
+            underlyingTokenIn = uniswapRouter.WETH();
```

Comment:
should initialize weth as constant or immutable to save gas and not repeatedly to call the view function